contracts/pcv/morpho/MorphoCompoundPCVDeposit.sol

```diff
@@ -66,54 +129,256 @@ contract MorphoCompoundPCVDeposit is PCVDeposit {
+            IPCVOracle(pcvOracle).updateLiquidBalance(
+                endingRecordedBalance - startingRecordedBalance
+            );
+        }
+    }
+
+    /// @notice set the pcv oracle address
+    /// @param _pcvOracle new pcv oracle to reference
+    function setPCVOracle(address _pcvOracle) external onlyGovernor {
+        address oldOracle = pcvOracle;
+        pcvOracle = _pcvOracle;
+
+        _recordPNL();
+
+        IPCVOracle(pcvOracle).updateLiquidBalance(
+            lastRecordedBalance.toInt256()
+        );
+
+        emit PCVOracleUpdated(oldOracle, _pcvOracle);
     }
 
+    /// ------------------------------------------
+    /// ------------- Helper Methods -------------
+    /// ------------------------------------------
+
     /// @notice helper function to avoid repeated code in withdraw and withdrawAll
-    function _withdraw(address to, uint256 amount) private {
-        MORPHO.withdraw(cToken, amount);
+    /// anytime this function is called it is by an external function in this smart contract
+    /// with a reentrancy guard. This ensures lastRecordedBalance never desynchronizes.
+    /// Morpho is assumed to be a loss-less venue. over the course of less than 1 block,
+    /// it is possible to lose funds. However, after 1 block, deposits are expected to always
+    /// be in profit at least with current interest rates around 0.8% natively on Compound,
+    /// ignoring all COMP and Morpho rewards.
+    /// @param to recipient of withdraw funds
+    /// @param amount to withdraw
+    /// @param recordPnl whether or not to record PnL. Set to false in withdrawAll
+    /// as the function _recordPNL() is already called before _withdraw
+    function _withdraw(
+        address to,
+        uint256 amount,
+        bool recordPnl
+    ) private {
+        /// ------ Effects ------
+
+        if (recordPnl) {
+            /// compute profit from interest accrued and emit a Harvest event
+            _recordPNL();
+        }
+
+        /// update last recorded balance amount
+        /// if more than is owned is withdrawn, this line will revert
+        /// this line of code is both a check, and an effect
+        lastRecordedBalance -= amount;
+
+        /// ------ Interactions ------
+
+        IMorpho(morpho).withdraw(cToken, amount);
         IERC20(token).safeTransfer(to, amount);
 
         emit Withdrawal(msg.sender, to, amount);
     }
 
-    /// @notice claim COMP rewards for supplying to Morpho
-    function harvest() external {
-        address[] memory cTokens = new address[](1);
-        cTokens[0] = cToken;
+    /// @notice records how much profit or loss has been accrued
+    /// since the last call and emits an event with all profit or loss received.
+    /// Updates the lastRecordedBalance to include all realized profits or losses.
+    function _recordPNL() private {
+        /// first accrue interest in Compound and Morpho
+        IMorpho(morpho).updateP2PIndexes(cToken);
```

Comment:
When calling withdraw or supply on Morpho the indexes are updated and this call might not be necessary.

Reply:
The issue is, we want to be able to know what the pcv deposit balance is including all unrealized gains or losses before depositing or withdrawing. To handle this, we must update the indexes first, which then means that when supply or withdraw is called on morpho, the interest indexes in morpho and compound don't get updated again.